Make policy attachments depend on users

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the policy attachment resources to depend on the `aws_iam_user.users` resource.

## 💭 Motivation and context ##

Resolves #22.

## 🧪 Testing ##

I successfully applied these changes to add a new user and verified that the failure I previously observer no longer happens.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

jsf9k committed:
Make policy attachments depend on users

I recently added a user, and when I ran terraform apply it failed
because aws_iam_user.users["new.user"] did not yet exist.  To fix
this, I made the policy attachments depend on the aws_iam_user.users
resource.  This forces all the users to be created before attempting
to attach policies to them.

dav3r reviewed (approved):
Good fix - makes sense to me. 🛫

mcdonnnj commented on users.tf:
```diff
@@ -12,6 +12,9 @@ resource "aws_iam_user" "users" {
 resource "aws_iam_user_policy_attachment" "self_managed_creds_with_mfa" {
   provider = aws.users
 
+  # Ensure that the user exists before attempting to attach the policy
```
```suggestion
  # Ensure that the users exist before attempting to attach the policy
```

jsf9k replied:
See commit dd57429.

mcdonnnj commented on users.tf:
```diff
@@ -23,6 +26,9 @@ resource "aws_iam_user_policy_attachment" "self_managed_creds_with_mfa" {
 resource "aws_iam_user_policy_attachment" "self_managed_creds_without_mfa" {
   provider = aws.users
 
+  # Ensure that the user exists before attempting to attach the policy
```
```suggestion
  # Ensure that the users exist before attempting to attach the policy
```

jsf9k replied:
See commit dd57429.

mcdonnnj reviewed (approved):
Thanks for resolving this! Just a commenting suggestion that you can take or leave.

jsf9k committed:
Explain why depends_on includes all users instead of just one

This is an important point that was glossed over by the previous comment,
due to it being overly brief.

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj reviewed (approved):
Verbose! I love it! 📜🏆
